### ISSUE TITLE
Generator improvements

### ISSUE THREAD

**Issue by retnuh** [arianvp/dhall-kubernetes#91]
Hi there, we've been using the generator to generate dhall types/defaults/schemas for other k8s related projects, like argo-workflows and argo-events.

I made a few improvements to the generator:

- Add --skipDuplicates CLI flag and handler.  
  - This allows you to just skip types that have the same `ModelName` but different namespaces; the types are still emmitted but skipped when creating the top-level types/defaults/schema files, etc.  Default behaviour is to error at, like current behaviour.
  - Can see the change [here](https://github.com/EarnestResearch/dhall-kubernetes/commit/6e948125881506a533d81422b0781098c3174576)

- Add --prefixMap option for specifying external import roots
  - This allows us to generate files for other projects that depend on k8s types and have the generated files import the desired repo or path.
  - Can see the change [here](https://github.com/EarnestResearch/dhall-kubernetes/commit/8c664612e2a5c1ace111f40234312ca4e157f33f)
  - For example:
```
$ dhall-kubernetes-generator --skipDuplicates --prefixMap io.k8s=https://raw.githubusercontent.com/dhall-lang/dhall-kubernetes/master/types/. argo-events-swagger.json
```
Then generates files that look like this:
```
$ cat types/io.argoproj.common.EventContext.dhall
{ cloudEventsVersion :
    Text
, contentType :
    Text
, eventID :
    Text
, eventTime :
    https://raw.githubusercontent.com/dhall-lang/dhall-kubernetes/master/types/io.k8s.apimachinery.pkg.apis.meta.v1.MicroTime.dhall
, eventType :
    Text
, eventTypeVersion :
    Text
, extensions :
    List { mapKey : Text, mapValue : Text }
, schemaURL :
    ./io.argoproj.common.URI.dhall
, source :
    ./io.argoproj.common.URI.dhall
}
```

If there's any interest in merging these changes back upstream I'd be happy to open one or more PRs.

**Comment by Gabriella439** [comment]
@retnuh: Yeah, I'd be happy to merge your changes

**Comment by Gabriella439** [comment]
This is fixed by #92
